activerecord/test/cases/associations_test.rb

```diff
@@ -241,6 +241,17 @@ def test_has_many_association_from_a_model_with_query_constraints_different_from
     assert_equal(expected_comments.sort, comments.sort)
   end
 
+  def test_foreign_key_is_in_ignored_columns
+    assert_raises ArgumentError do
```

Comment:
You can use `match` to assert to expected ArgumentError is raised:
```suggestion
    assert_raises ArgumentError, match: /the column post_id is ignored/ do
```